hordes-ui-mod.js

```diff
@@ -382,941 +392,1071 @@
+            // Simulating clicks on the filters to turn them on/off
+            const $filterButtons = Array.from(document.querySelectorAll('.channelselect small'));
+            Object.keys(tabFilters).forEach(filter => {
+                const $filterButton = $filterButtons.find($btn => $btn.textContent === filter);
+                const isCurrentlyFiltered = $filterButton.classList.contains('textgrey');
+
+                // If is currently filtered but filter for this tab is turned off, click it to turn filter off
+                if (isCurrentlyFiltered && !tabFilters[filter]) {
+                    $filterButton.click();
+                }
+                // If it is not currently filtered but filter for this tab is turned on, click it to turn filter on
+                if (!isCurrentlyFiltered && tabFilters[filter]) {
+                    $filterButton.click();
+                }
+            });
+
+            // Update state for our custom chat filters to match the tab's configuration, then filter chat for it
+            const isGMChatVisible = !tabFilters.GM;
+            modHelpers.setGMChatVisibility(isGMChatVisible);
+
+            // Update the selected tab in state
+            state.selectedChatTabId = tabId;
+            save();
+        },
+
+        // Updates state.chat.GM and the DOM to make text white/grey depending on if gm chat is visible/filtered
+        // Then filters chat and saves updated chat state
+        setGMChatVisibility: isGMChatVisible => {
+            const $chatGM = document.querySelector(`.js-chat-gm`);
+            state.chat.GM = isGMChatVisible;
+            $chatGM.classList.toggle('textgrey', !state.chat.GM);
+            modHelpers.filterAllChat();
+            save();
+        },
+
+        /** get char level */
+        getCurrentLvl: () => Number(document.querySelector('#ufplayer > div > div > div.progressBar.bgmana > .left').textContent.split('Lv. ')[1]),
+
+        /** get current xp */
+        getCurrentXp: () => Number(document.querySelector('#expbar > .bar > .progressBar > .left').textContent.split('/')[0].trim()),
+
+        /** get next level xp */
+        getNextLevelXp: () => Number(document.querySelector('#expbar > .bar > .progressBar > .left').textContent.split('/')[1].trim()),
+
+        /** user invoked reset of xp meter stats */
+        resetXpMeterState: () => {
+            state.xpMeterState.xpArr = [];
+            state.xpMeterState.averageXp = 0;
+            state.xpMeterState.gainedXp = 0;
+            document.querySelector('#timeremain').textContent = '-:-:-'
+        },
+
+        /** toggle the xp meter */
+        toggleXpMeter: () => {
+            const xpMeterContainer = document.querySelector('#xpmeter')
+            xpMeterContainer.style.display === "none" ? xpMeterContainer.style.display = "block" : xpMeterContainer.style.display = "none";
+        },
+
+        msToString: (ms) => {
```

Comment:
Could you move this function to `utils` near the bottom of the file?
It's not directly related to/reliant on any of the specific mods. I've been trying to keep these types of methods separate in utils, if that makes some sense.